dev/src/DocGenerator/Parser/CodeParser.php

```diff
@@ -953,6 +953,36 @@ private function getSource($path)
         return implode('/', array_slice($filePieces, $base));
     }
 
+    private function getClassReference($type)
+    {
+        // Depending on the currently used PHP version, the following classes cannot
+        // reliably be reflected. PHP 5.x doesn't support return type hints, some
+        // third party dependencies require type hints and throw a Parse error
+        // when an implementation is not compatible with its parent class or
+        // interface.
+        static $map = [
+            // Monolog 1.x/2.x related classes
+            'Google\Cloud\Core\Logger\AppEngineFlexFormatter' => 'Core/src/Logger/AppEngineFlexFormatter.php',
+            'Google\Cloud\Core\Logger\AppEngineFlexFormatterV2' => 'Core/src/Logger/AppEngineFlexFormatter.php',
```

Comment:
Instead of doing this I would oriented here how Symfony implements bridges between version example: https://github.com/symfony/phpunit-bridge/blob/master/SymfonyTestsListener.php#L14-L20

For the monolog version we can check for: `Monolog\Logger::API` version and implement.

So we create a `AppEngineFlexFormatterV2` and a `AppEngineFlexFormatterV1` and based on `AppEngineFlexFormatter.php` would look like this:

```php
namespace Google\Cloud\Core\Logger;

if (version_compare(\Monolog\Logger\::API, '2.0.0', '<')) {
    class_alias('Google\Cloud\Core\Logger\AppEngineFlexFormatterV1', 'Google\Cloud\Core\Logger\AppEngineFlexFormatter');
} else {
    class_alias('Google\Cloud\Core\Logger\AppEngineFlexFormatterV2', 'Google\Cloud\Core\Logger\AppEngineFlexFormatter');
}

if (false) {
    class AppEngineFlexFormatter
    {
    }
}
```

Reply:
@jeromegamez why not use class aliases instead of the AppEngineFlexHandlerFactory?

Reply:
Sorry for the unformatted reply, I‘m currently on the road. I was glad that I got it working in the end at all, and the Factory is similar to the Guzzle HttpHandlerFacory in the google/auth library so it‘s a „known procedure“.

Class aliases in the scope of a testing framework like the symfony phpunit Bridge make sense to me, but I’m not sure it’s the most transparent approach for direct use here. I haven’t tested it, but I also don’t know if it works well with the different PHP versions (5.6 vs. 7.x).

But that‘s just my thoughts right now, I would like for John to decide this - if he gives a „go“ I‘ll happily implement it.

Reply:
@jdpedrie Yeah the comment is a little bit missplayed and its ony a input from my side, I'm ok with the current implementation as it seems to be also a common pattern used not only here.